distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/CometDClient.java

```diff
@@ -0,0 +1,199 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.cometd.bayeux.Channel;
+import org.cometd.bayeux.Message;
+import org.cometd.bayeux.client.ClientSessionChannel;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.ClientTransport;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+import org.springframework.util.CollectionUtils;
+
+public class CometDClient {
+
+    private static final XLogger LOGGER = new XLogger(LoggerFactory.getLogger(CometDClient.class));
+    
+    private static final long TIMEOUT = TimeUnit.SECONDS.toMillis(60);
+    
+    private final BayeuxClient bayeuxClient;
+
+    private final HttpClient httpClient;
+
+    private final List<MessageListener> messageListeners = new ArrayList<>();
+
+    public CometDClient(String url) throws Exception {
+        SslContextFactory sslContextFactory = new SslContextFactory(true);
+        httpClient = new HttpClient(sslContextFactory);
+        doTrustAllCertificates();
+        ClientTransport transport = new LongPollingTransport(new HashMap<>(), httpClient);
+        bayeuxClient = new BayeuxClient(url, transport);
+    }
+
+    public boolean start() throws Exception {
+        httpClient.start();
+        LOGGER.debug("HTTP client started: {}", httpClient.isStarted());
+        MessageListener handshakeListener = new MessageListener(Channel.META_HANDSHAKE);
+        bayeuxClient.getChannel(Channel.META_HANDSHAKE).addListener(handshakeListener);
+        bayeuxClient.handshake();
+        boolean connected = bayeuxClient.waitFor(TIMEOUT, BayeuxClient.State.CONNECTED);
+        return connected;
+    }
+
+    public void subscribe(String channel) {
+        if (bayeuxClient.isConnected()) {
+            if (!alreadySubscribed(channel)) {
+                ClientSessionChannel clientSessionChannel = bayeuxClient.getChannel(channel);
+                MessageListener messageListener = new MessageListener(channel);
+                clientSessionChannel.subscribe(messageListener);
+                messageListeners.add(messageListener);
+            } else {
+                LOGGER.warn("Already subscribed to channel {}", channel);
+            }
+        } else {
```

Comment:
Just for consistency sake, maybe all methods should first check the connection state and/or channel subscription, throw an exception if something's wrong and skip the `else` statement?